src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -477,6 +478,21 @@ public final class Guardrails implements GuardrailsMBean
                      format("The modification to table %s has a timestamp %s before the minimum allowable %s threshold %s",
                             what, value, isWarning ? "warning" : "failure", threshold));
 
+    public static final MaxThreshold maximumReferencedIndexesWithNonParitionRestrictedQuery =
+    new MaxThreshold("maximum_referenced_indexes_for_non_paritition_key_restricted_query",
+                     "High number of referenced indexes per query may negatively effect performance",
+                     state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReferencedIndexesWithNonPartitionRestrictedQueryWarnThreshold(),
+                     state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReferencedIndexesWithNonPartitionRestrictedQueryFailThreshold(),
+                     ((isWarning, what, value, threshold) ->
+                      format("The number of referenced indexes on a table with %s on index %s violated %s threshold value %s with value %s",
+                             LeveledCompactionStrategy.class.getName(), what, isWarning ? "warning" : "failure", threshold, value)));
+
+    public static final EnableFlag nonPartitionRestrictedQuery =
+    new EnableFlag("non_partition_restricted_query",
```

Comment:
nit: Make sure naming is consistent at the end w/ what's in the YAML and `Config`